docs/projects/run-build-deploy.md

```diff
@@ -147,20 +161,90 @@ image that was set when the function was added to the project.
+image. In addition to that, a project can contain the configurations for building this default image. This is done 
+through the {py:meth}`~mlrun.projects.MlrunProject.build_config` and {py:meth}`~mlrun.projects.MlrunProject.build_image` 
+functions. 
+
+The project build configuration is maintained in the project object. When saving, exporting and importing the project 
+these configurations are carried over with it. This makes it simple to transport a project between systems while 
+ensuring the needed runtime images are built and are ready for execution. 
+
+When using {py:meth}`~mlrun.projects.MlrunProject.build_config` build configurations can be passed along with the 
+resulting image name, and these are used to build the image. The image name is assigned following these rules:
+
+1. The name passed in the `image` parameter of the {py:meth}`~mlrun.projects.MlrunProject.build_config` function.
+2. The project's default image name, if configured using {py:meth}`~mlrun.projects.MlrunProject.set_default_image`.
+3. The value set in MLRun's `default_project_image_name` config parameter - by default this value is 
+   `.mlrun-project-image-{name}` with the project name as template parameter.
+
+For example:
+
+```python
+ # Set image config for current project object, using base mlrun image with additional requirements. 
+ image_name = ".my-project-image"
+ project.build_config(
+     image=image_name,
+     set_as_default=True,
+     with_mlrun=False,
+     base_image="mlrun/mlrun",
+     requirements=["vaderSentiment"],
+ )
+
+ # Export project configuration. The yaml file will contain build configuration in it
+ proj_file_path = "~/mlrun/my-project/project.yaml"
+ project.export(proj_file_path)
+```
+ 
+This project can then be imported and the default image can be built:
+
+```python
+ # Import the project as a new project with a different name
+ new_project = mlrun.load_project("~/mlrun/my-project", name="my-other-project")
+ # Build the default image for the project, based on project build config
+ new_project.build_image()
 
-    # function will run with the "myrepo/my-prebuilt-image:v1" image
-    run1 = project.run_function("prep", params={"x": 7}, inputs={'data': data_url})
+ # Set a new function and run it, will use the my-project-image image built previously
+ new_project.set_function("sentiment.py", name="scores", kind="job", handler="handler")
+ new_project.run_function("scores")
+```
+
+<a id="build_image"></a>
+## build_image
 
-    ...
+The {py:meth}`~mlrun.projects.MlrunProject.build_image` function builds an image using the existing build configuration, 
+and it can also be used to set the build configuration and build the image based on it in a single step. 
+When using `set_as_default=False` any build config provided will still be kept in the project object
+but the generated image name will not be set as the default image for this project. 
```

Comment:
When using `set_as_default=False` any build config provided is still be kept in the project object
but the generated image name is not set as the default image for this project.